### PR TITLE
Record keyframe animation as GIF and MP4

### PR DESCRIPTION
**Pull request by m-agour**
This PR introduces a micro feature which is the ability to record the animation as GIF and MP4.

An example of GIF record:
arm

MP4 Record:

https://user-images.githubusercontent.com/63170874/190893057-1cffac89-f560-4817-af53-df440e892537.mp4

### CODE REVIEW

File: fury/animation/tests/test_timeline.py [hunk 1]
```diff
@@ -102,3 +103,15 @@ def test_timeline():
                                 cube.GetScale())
         npt.assert_almost_equal(tl.get_rotation(tl.current_timestamp),
                                 cube.GetOrientation())
+
+    scene = Scene()
+    showm = ShowManager(scene)
+    scene.add(tl)
+    scene.GetActiveCamera().SetPosition(0, 0, -100)
```
Comment (skoudoro) [inline fury/animation/tests/test_timeline.py:110]: scene.set_camera(position=(0,0,-100))

File: fury/animation/timeline.py [hunk 2]
```diff
@@ -1495,6 +1497,100 @@ def has_playback_panel(self):
...
+            The file name. Save a GIF file if name ends with '.gif', or mp4
+            video if name ends with'.mp4'.
+            If None, this method will only return an array of frames.
+        fps : int, optional, default 30
+            The number of frames per second of the record.
+        size : (int, int)
+            ``(width, height)`` of the window. Default is (900, 768).
+        speed : float, optional, default 1.0
+            The speed of the animation.
+        order_transparent : bool
```
Comment (skoudoro) [inline fury/animation/timeline.py:1517]: `, optional`

**Review by skoudoro — CHANGES_REQUESTED** [review]
Hi @m-agour,

Let's tackle this PR! First, it needs to be rebased and conflicts needs to be fixed. 

Can you also remind me what else needs to be done in this PR?

Let me know when I can review it. Thanks

**Comment by skoudoro** [comment]
Do you have an update on this @m-agour ?

**Comment by m-agour** [comment]
@skoudoro I am working on it. Recording should be fast, but it currently takes more time than it should. I am working on speeding it up by minimizing or eliminating new VTK objects being created inside the recording loop.

**Comment by skoudoro** [comment]
thank you for the feedback !

**Comment by Garyfallidis** [comment]
self.window_to_image_filter.Modified()
self.window_to_image_filter.Update()

**Comment by skoudoro** [comment]
Hi @m-agour,

Do you plan to do the update here? What is your timeline? Thank you for your feedback!

**Comment by skoudoro** [comment]
ping @m-agour

**Comment by m-agour** [comment]
@skoudoro Sorry for being behind schedule. I had some troubles trying to multi-thread the recording process. By tomorrow I would finish it and ping you

**Comment by m-agour** [comment]
@skoudoro I modified the record method as we discussed by eliminating the creation of `WindowToImageFilter()` inside the rendering loop. Also eliminated all unnecessary loops except for the Rendering one. As for multi-threading but it has a lot of issues. So, I was thinking we could merge this and approach the multithreading option after.

**Comment by skoudoro** [comment]
I agree for the multithreading approach in a new PR. 

I will test and review this PR a bit later today. thank you @m-agour !

**Pull request merged**